mx3_beamline_library/devices/classes/robot.py

```diff
@@ -197,6 +197,13 @@ def _set_and_wait(self, value=None, timeout: float = None) -> bytes:
         bytes
             The robot response
         """
+        # Check if the robot has pre-picked a pin
+        if self.client.status.state.jaw_a_pin is not None:
+            self.client.trajectory.puck.return_pin()
+            # Wait until operation is complete
+            sleep(0.5)
+            while self.client.status.state.path != RobotPaths.UNDEFINED:
+                sleep(0.5)
```

Comment:
Should be able to pass `wait=True` to achieve the same effect, since `PuckTraj.return_pin` is wrapped by the `wait_for_path` decorator.

e.g.
```python
if self.client.status.state.jaw_a_pin is not None:
    self.client.trajectory.puck.return_pin(wait=True)
```

See
https://github.com/AustralianSynchrotron/mx-robot-library/blob/main/mx_robot_library/trajectory/puck.py#L289
And
https://github.com/AustralianSynchrotron/mx-robot-library/blob/0cc66afc9a6e2e7872c14042082d827ed1b28e60/mx_robot_library/decorators.py#L341